### ISSUE TITLE
Usage in Gatsby 2

### ISSUE THREAD

**Issue by timsuchanek** [silvenon/gatsby-transformer-mdx#1]
This is an awesome project! Thanks for creating this.

I see that you're exposing both `jsx` and `mdxHast`.

How can that be rendered in the React frontend?
As we can't use the `dangerouslySetInnerHTML` and need to execute the imports from the mdx source.

Would be awesome if you could show a small example how you imagine that to be used.

Thanks!

**Comment by timsuchanek** [comment]
I just saw that in  https://github.com/gatsbyjs/gatsby/issues/312 there is a solution to this.

**Comment by silvenon** [comment]
Honestly I don’t know how to make it work properly in Gatsby v2. This transformer was my attempt to do it without the loader, but I don’t know what to expose in GraphQL to be able to render it. Any ideas would be highly appreciated.

**Comment by timsuchanek** [comment]
What is the major difference that makes it hard with Gatsby v2?

**Comment by silvenon** [comment]
The existing solutions work in Gatsby v1, this is a WIP attempt at making it work in v2. In v1 you can [set a layout](https://github.com/avigoldman/avigoldman.com/blob/master/gatsby-node.js), this doesn't exist in v2. I don't know how to get around that, at least not without being pretty repetitive.

**Comment by kwangkim** [comment]
avigoldman actually used `gatsby-plugin-mdx` which contains a mdx-loader. You may need a loader.

**Comment by silvenon** [comment]
Yeah, I’ll definitely need a loader, I found that out a while later. I’ll probably dump this project and encourage avigoldman to publish his, since it was his idea anyway.
